apps/maestro/src/features/InterchainTokenDeployment/steps/deploy-and-register/DeployAndRegister.tsx

```diff
@@ -100,6 +101,38 @@ export const Step2: FC = () => {
       });
 
       const txPromise = deployInterchainTokenAsync();
+      console.log("rootState.selectedChains", rootState.selectedChains);
+
+      // Sui will return a digest equivalent to the txHash
+      const SUI_CHAIN_ID = NEXT_PUBLIC_NETWORK_ENV === "mainnet" ? 101 : 103;
```

Comment:
why is the Sui handler that much more complex than the EVM handler in this case? the EVM handler seems to only setTxState to idle, whereasyou're also invoking an addTransaction function here for SUi